modmail/extensions/extension_manager.py

```diff
@@ -0,0 +1,301 @@
+        self.bot = bot
+        self.all_extensions = EXTENSIONS
+        self.refresh_method = walk_extensions
+
+    def get_black_listed_extensions(self) -> list:
+        """Returns a list of all unload blacklisted extensions."""
+        return NO_UNLOAD
+
+    @commands.group("ext", aliases=("extensions", "exts"), invoke_without_command=True)
+    async def extensions_group(self, ctx: Context) -> None:
+        """Load, unload, reload, and list loaded extensions."""
+        await ctx.send_help(ctx.command)
+
+    @extensions_group.command(name="load", aliases=("l",))
+    async def load_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Load extensions given their fully qualified or unqualified names.
+
+        If '\*' or '\*\*' is given as the name, all unloaded extensions will be loaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        if "*" in extensions or "**" in extensions:
+            extensions = sorted(ext for ext in self.all_extensions if ext not in self.bot.extensions.keys())
+
+        msg = self.batch_manage(Action.LOAD, *extensions)
+        await ctx.send(msg)
+
+    @extensions_group.command(name="unload", aliases=("ul",))
+    async def unload_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Unload currently loaded extensions given their fully qualified or unqualified names.
+
+        If '\*' or '\*\*' is given as the name, all loaded extensions will be unloaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        blacklisted = [ext for ext in self.get_black_listed_extensions() if ext in extensions]
+
+        if blacklisted:
+            bl_msg = "\n".join(blacklisted)
+            await ctx.send(f":x: The following {self.type}(s) may not be unloaded:```\n{bl_msg}```")
+            return
+
+        if "*" in extensions or "**" in extensions:
+            extensions = sorted(
+                ext
+                for ext in self.bot.extensions.keys() & self.all_extensions
+                if ext not in (self.get_black_listed_extensions())
+            )
+
+        await ctx.send(self.batch_manage(Action.UNLOAD, *extensions))
+
+    @extensions_group.command(name="reload", aliases=("r", "rl"))
+    async def reload_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Reload extensions given their fully qualified or unqualified names.
+
+        If an extension fails to be reloaded, it will be rolled-back to the prior working state.
+
+        If '\*' or '\*\*' is given as the name, all currently loaded extensions will be reloaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        if "*" in extensions or "**" in extensions:
```

Comment:
Ditto. Don't forget the docstring

Reply:
And everywhere else this is. It's in a lot of places. We should make this an atomic commit too, in case we go back and make ** mean something again.